### PR TITLE
Install Compass as a separate component

### PR DESCRIPTION
**Pull request by kjaksik**
**Description**

This PR aims to install Compass as a separate component from Kyma, with seprate Installation CR and Operator

Changes proposed in this pull request:

- sync compass chart with the one from Kyma repo
- update `run.sh` so that it installs Compass with a seprate operator
- introduce `KYMA_VERSION` file with the version of Kyma that will be used during the installation

**Related issue(s)**

### CODE REVIEW

File: chart/compass/charts/cockpit/OWNERS [hunk 1]
```diff
@@ -1,7 +1,10 @@
+reviewers:
+  - console-reviewers
```
Comment (pkosiec) [inline chart/compass/charts/cockpit/OWNERS:2]: Please also copy OWNERS ALIASES

File: installation/cmd/run.sh [hunk 2]
```diff
@@ -3,34 +3,37 @@
...
+bash "${ROOT_PATH}"/installation/scripts/run-compass-installer.sh
+bash "${ROOT_PATH}"/installation/scripts/is-installed.sh
+#MINIKUBE_IP=$(eval minikube ip)
+#helm install \
+#-f=${MINIKUBE_HELM_VALUES} \
+#--set=global.minikubeIP=${MINIKUBE_IP} \
+#--name "${COMPASS_HELM_RELEASE_NAME}" \
+#--namespace "${COMPASS_HELM_RELEASE_NAMESPACE}" \
+#"${ROOT_PATH}"/chart/compass \
+#--tls --wait
```
Comment (pkosiec) [inline installation/cmd/run.sh:34]: Remove commented code

File: installation/cmd/run.sh [hunk 3]
```diff
@@ -3,34 +3,37 @@
...
+
```
Comment (pkosiec) [inline installation/cmd/run.sh:35]: Please remove minikube values

File: chart/compass/charts/connectivity-adapter/copy-from-compass.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (pkosiec) [inline chart/compass/charts/connectivity-adapter/copy-from-compass.sh:1]: Remove it please

File: chart/compass/charts/connector/copy-from-compass.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+## This script copies charts from Compass repository
```
Comment (pkosiec) [inline chart/compass/charts/connector/copy-from-compass.sh:3]: Remove it please

File: chart/compass/charts/director/templates/service.yaml [hunk 6]
```diff
@@ -19,7 +19,7 @@ spec:
     app: {{ .Chart.Name }}
     release: {{ .Release.Name }}
 ---
-{{- if eq .Values.global.metrics.enabled true -}}
+{{- if and .Values.global.metrics.enabled -}}
```
Comment (pkosiec) [inline chart/compass/charts/director/templates/service.yaml:22]: Revert this line

File: chart/compass/charts/director/templates/service-monitor.yaml [hunk 7]
```diff
@@ -1,4 +1,4 @@
-{{- if eq .Values.global.metrics.enabled true -}}
+{{- if and .Values.global.metrics.enabled -}}
```
Comment (pkosiec) [inline chart/compass/charts/director/templates/service-monitor.yaml:1]: Revert this line

File: chart/compass/charts/metris/values.yaml [hunk 8]
```diff
@@ -38,12 +38,12 @@ provider:
...
-    namespace: "kyma-dev"
+    name: "TBD"
+    namespace: "TBD"
     version: 1
-    env: "dev"
-  token: ""
-  url: "https://input.yevents.io"
+    env: "TBD"
+  token: "TBD"
+  url: "TBD"
```
Comment (pkosiec) [inline chart/compass/charts/metris/values.yaml:46]: I think it should be reverted. Correct me if I'm wrong, @maksd

File: chart/compass/charts/provisioner/copy-from-compass.sh [hunk 9]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (pkosiec) [inline chart/compass/charts/provisioner/copy-from-compass.sh:1]: Remove this script

File: installation/cmd/run.sh [hunk 10]
```diff
@@ -3,34 +3,27 @@
...
--f=${MINIKUBE_HELM_VALUES} \
---set=global.minikubeIP=${MINIKUBE_IP} \
---name "${COMPASS_HELM_RELEASE_NAME}" \
---namespace "${COMPASS_HELM_RELEASE_NAMESPACE}" \
-"${ROOT_PATH}"/chart/compass \
---tls --wait
+bash "${ROOT_PATH}"/installation/scripts/run-compass-installer.sh
+bash "${ROOT_PATH}"/installation/scripts/is-installed.sh
 
 # TODO: Remove it after next CLI release
```
Comment (tgorgol) [inline installation/cmd/run.sh:28]: Seems like CLI [adds](https://github.com/kyma-project/cli/blob/47ab12cc176b30dffe8ae2c0abd7e174b5c3c8ad/cmd/kyma/install/cmd.go#L224) to hosts all virtual services now, maybe we could remove this already? :)

File: installation/resources/installer-config-local.yaml.tpl [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
...W9UQ0llOVpxN2Q3U0ZXckZZeW1UdVZOUFByZlo1bHhwOGphTTRVbTd4MnpReGJ2UERHb3o1YXdHV0wKRThGNncwaEF1UzZValVJazBLbE9vamVxQnh3L1JBcGNrUTNlTXNXbEQwNENTb2tyNFJhWlBmVllrY2ZBWWNaWgpOdWR6ZjBKMC9GU0ZTbjN4L0RoNTROV2NGS1IxUnpBVGVaVUJ4cVZSQW9JQkFRRFlDNmZvVWpOQnJ2ckNFVzkrCkhYZlk1Ni9CbUZ4U3hUTHU0U2h6VncwakViZTltVWljQ2pDc1hQMUwySVJCdEdaWU9YWTVqdDlvSzlSV0RSdVMKWUZqZFdmemduU1lWRmZyZUw0emRQVGlxbGEvQjhMNWptVlNoeGNycmxheE02Uk1FWjFlZGtDa1ZPbTFQdmwzVAo1TW5OZGhySXFWeE1OMWxjRVdiU29vclJpUW9Lb3poMHRQSG9YckZBbG9BZVJ3bHpWeE9jb21ZVzJiaDBHUzdmCjVoaHZoZWUxYmVISnY3UXFoWkU3WUhxSU9iTVBaUWJqWEdnRkxmMnlDRitzM2Jtem1DRFJTN0V6ZVdxSXVDdVMKTlZUYU0rSzZyQlRoN0NLRjZUWlNqQW55SmZoRmRlT1ZKNzlNZDEzYWVJaG0zNTB6UWc3dWZKL2drdkorNUR2TApacC9uQW9JQkFBVlg4WHpFTzdMVk1sbENKZFVUdURTdXNPcDNrQVlFZ2dZNFFRM3FNTlcxRnl5WEM3WjBGOWJFCmtTSEhkalJtU2RUbFZueGN2UW1KTS9WL2tJanpNUHhFT3NCS1BVVkR3N3BhOHdiejlGcTRPOCtJb3lqN1ZXclcKMmExL1FNWXlzSGlpTlBzNy8vWUtvMy9rdkhCWUY2SnNkenkyQkVSTkQ0aTlVOWhDN0RqcGxKR3BSNktMTVBsegpNWFJ3VjVTM2V3cnBXZVcxQW5ONC9EKy9zUGlNQTNnS0swSlBFdGVBV1dndEZHTnNBSkJnaFBoUExxQi9CcDUvCkhOeC96M0w0MWtqRnpqOHNWaHMrVDRZYlhiaGF2R2xxc2h5ZldQbnRhV1VOMG15MjU1RFdQUDhWa24yeFNlV2kKVm1hVW5TSDBTZ2tlUENMRnlra25yQzgxU2pXZkRBMD0KLS0tLS1FTkQgUFJJVkFURSBLRVktLS0tLQo="
```
Comment (tgorgol) [inline installation/resources/installer-config-local.yaml.tpl:24]: Can't see those values used in any chart, maybe we could get rid of it?

File: installation/scripts/build-kyma-installer.sh [hunk 12]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+echo "The build-kyma-installer.sh script is deprecated and will be removed. Use Kyma CLI instead."
```
Comment (tgorgol) [inline installation/scripts/build-kyma-installer.sh:5]: I wonder if we should delete those warnings, in our case Kyma CLI for this is not an option yet.

File: installation/scripts/build-kyma-installer.sh [hunk 13]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
```
Comment (tgorgol) [inline installation/scripts/build-kyma-installer.sh:1]: Should we rename this script to `build-compass-installer.sh`?

File: installation/scripts/build-kyma-installer.sh [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+echo "The build-kyma-installer.sh script is deprecated and will be removed. Use Kyma CLI instead."
+
+CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR=${CURRENT_DIR}/../../
+IMAGE_NAME="$(${CURRENT_DIR}/extract-compass-installer-image.sh)"
+BUILD_ARG=""
+
+echo "
+################################################################################
+# Kyma-Installer build
```
Comment (tgorgol) [inline installation/scripts/build-kyma-installer.sh:14]:
```suggestion
# Compass-Installer build
```

File: installation/scripts/configure-components.sh [hunk 15]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+echo "The configure-components.sh script is deprecated and will be removed. Use Kyma CLI instead."
```
Comment (tgorgol) [inline installation/scripts/configure-components.sh:5]: I wonder if we should delete those warnings, in our case Kyma CLI for this is not an option yet.

File: installation/scripts/configure-components.sh [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+readonly CM_NAME="kyma-sub-components"
+readonly LABEL="installer=overrides"
+
+# Do nothing if the components.env is empty or does not exist at all
+if [[ ! -f "${FILE_PATH}" || ! -s "${FILE_PATH}" ]]; then
+    exit 0
+fi
+
+kubectl create configmap ${CM_NAME} -n kyma-installer --from-env-file="${FILE_PATH}" --dry-run -o yaml | kubectl apply -f -
+kubectl label configmap ${CM_NAME} -n kyma-installer ${LABEL}
```
Comment (tgorgol) [inline installation/scripts/configure-components.sh:20]:
```suggestion
kubectl create configmap ${CM_NAME} -n compass-installer --from-env-file="${FILE_PATH}" --dry-run -o yaml | kubectl apply -f -
kubectl label configmap ${CM_NAME} -n compass-installer ${LABEL}
```

File: installation/scripts/install-tiller.sh [hunk 17]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+echo "The install-tiller.sh script is deprecated and will be removed. Use Kyma CLI instead."
```
Comment (tgorgol) [inline installation/scripts/install-tiller.sh:3]: I wonder if we should delete those warnings, in our case Kyma CLI for this is not an option yet.

File: installation/scripts/installer.sh [hunk 18]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+echo "The installer.sh script is deprecated and will be removed. Use Kyma CLI instead."
```
Comment (tgorgol) [inline installation/scripts/installer.sh:5]: I wonder if we should delete those warnings, in our case Kyma CLI for this is not an option yet.

File: installation/scripts/installer.sh [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+            ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+echo "
+################################################################################
+# Kyma Installer setup
```
Comment (tgorgol) [inline installation/scripts/installer.sh:47]:
```suggestion
# Compass Installer setup
```

File: installation/scripts/run-compass-installer.sh [hunk 20]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+echo "The run.sh script is deprecated and will be removed. Use Kyma CLI instead."
```
Comment (tgorgol) [inline installation/scripts/run-compass-installer.sh:5]: I wonder if we should delete those warnings, in our case Kyma CLI for this is not an option yet.

File: tools/compass-installer/compass.Dockerfile [hunk 21]
```diff
@@ -0,0 +1,7 @@
+ARG INSTALLER_VERSION="PR-8638"
```
Comment (tgorgol) [inline tools/compass-installer/compass.Dockerfile:1]: Maybe we could add a comment to this ARG, saying that this is the version of Kyma Operator (from Kyma repository) that will be used as a base for Compass Installer?

File: tools/compass-installer/compass.Dockerfile [hunk 22]
```diff
@@ -0,0 +1,8 @@
...
+# INSTALLER_VERSION is a the base version of the Kyme Operator that will be used to build Compass Installer
```
Comment (pkosiec) [inline tools/compass-installer/compass.Dockerfile]: Please move this comment to top

File: tools/compass-installer/compass.Dockerfile [hunk 23]
```diff
@@ -1,3 +1,4 @@
+# The base version of the Kyme Operator that will be used to build Compass Installer
```
Comment (tgorgol) [inline tools/compass-installer/compass.Dockerfile:1]:
```suggestion
# The base version of the Kyma Operator that will be used to build Compass Installer
```

**Comment by pkosiec** [comment]
Yup, as @tgorgol said, other codeowners need to verify their charts, to make sure all's up to date.

File: chart/compass/charts/metris/values.yaml (hunk 8 above)
Comment (maksd) [reply chart/compass/charts/metris/values.yaml:46]: @pkosiec it's OK, we can leave it like that.

File: chart/compass/values.yaml [hunk 24]
```diff
@@ -109,20 +111,20 @@ global:
     # If secrets do not exist they will be created
     secrets:
       ca:
-        name: compass-connector-app-ca
-        namespace: compass-system
+        name: application-connector-certificate-overrides
+        namespace: compass-installer
```
Comment (Szymongib) [inline chart/compass/values.yaml:115]: I do not think that it is correct.

Comment (kjaksik) [reply chart/compass/values.yaml:115]: Should I revert it then? There's an override for this secret [here](https://github.com/kyma-incubator/compass/blob/96ceae69260e35234561995bcdb0c0b44ee182a7/installation/resources/installer-config-local.yaml.tpl#L4) if this changes anything.

File: chart/compass/templates/agent-configuration-job.yaml [hunk 25]
```diff
@@ -0,0 +1,223 @@
+{{ if .Values.global.agentPreconfiguration }}
```
[the remark this answers is not in the document]
Comment (kjaksik) [reply chart/compass/templates/agent-configuration-job.yaml:1]: Sure, will create it in a moment

[the remark this answers is not in the document]
Comment (kjaksik) [reply chart/compass/templates/agent-configuration-job.yaml:1]: https://github.com/kyma-incubator/compass/issues/1419

File: chart/compass/charts/kyma-environment-broker/templates/tests/e2e-upgrade-azure.yaml [hunk 26]
```diff
@@ -30,9 +30,9 @@ spec:
                 name: {{ include "kyma-env-broker.fullname" . }}
           env:
             - name: APP_PROVISION_TIMEOUT
-              value: "{{ .Values.e2e.provisioning.timeout }}"
+              value: "105m"
```
Comment (polskikiel) [inline chart/compass/charts/kyma-environment-broker/templates/tests/e2e-upgrade-azure.yaml:33]: you may revert changes against this file, it's outdated on the kyma side

File: chart/compass/charts/kyma-environment-broker/templates/tests/e2e-provisioning-gcp.yaml [hunk 27]
```diff
@@ -36,7 +36,7 @@ spec:
               # then this test will be executed on the pre-master-compass-integration job with the rest of the acceptance tests.
               # With the Dummy test, we won't block the pipelines.
               # In future we can make that test useful for pre-master jobs.
-              value: "true"
+              value: "false"
```
Comment (polskikiel) [inline chart/compass/charts/kyma-environment-broker/templates/tests/e2e-provisioning-gcp.yaml:39]: you may revert changes against this file, the reason is in the comment above.

**Pull request merged**